streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/TimeKey.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream.internals.suppress;
+
+import java.util.Objects;
+
+class TimeKey<K> implements Comparable<TimeKey<K>> {
```

Comment:
This is specifically for storing the keys sorted by timestamp in the buffer. I wasn't sure whether a more general or more specific name like `BufferKey` is better...

Reply:
Yeah, this sounds good.

Reply:
Hmm. Actually, Stamped has unusual implementations of equals, hashcode, and compareTo. They all disregard the stamped value and are only determined by the timestamp...

So, Stamped won't provide the semantics we need from TimeKey, and I'm afraid to change the equals/hashcode/compareTo of Stamped and messing up _its_ semantics...

WDYT?

Reply:
Do we need to require value ordering for `TimeKey` here? I thought it is not required as they are not following offset ordering to break ties anyways, right?

Reply:
I don't _think_ that will work...

`Comparable` requires a total ordering and also specifies that `a.compareTo(b) == 0` iff `a.equals(b)`, which in turn requires that `a.hashCode() == b.hashCode()`.

But this would prevent us from inserting two different keys with the same time into our buffer map. It doesn't seem like `Stamped` is suitable for map keys or set entries for this reason.